boa/src/syntax/lexer/number.rs

```diff
@@ -226,6 +226,16 @@ impl<R> Tokenizer<R> for NumberLiteral {
             }
         }
 
+        // Checks if the next char after '0x', '0o' or '0b' is a digit
+        // of that base. if not return an error. 
+        if let Some(digit) = cursor.peek()? {
+            if !digit.is_digit(kind.base()) {
+                return Err(Error::syntax(
+                    "expected digit after number base prefix",
+                    cursor.pos(),
+                ));
+            }
+        }
```

Comment:
I think this is missing checks that the base is 2/8/16 and also this should be moved up into the if (0) statement above

Reply:
> I think this is missing checks that the base is 2/8/16 and also this should be moved up into the if (0) statement above,

Probably best if we move this to the `0x`, `0o`, and `0b` cases this should also remove some check, what base it should be and and what base the digit should be.

so on line 158 (hexadecimal 0x) we peek a char and if it is not a digit with base `16` we return error "expected hexadecimal digit after '0x' prefix"

on line 166 (ocatal 0o) we peek a char and if it is not a digit with base `8` we return error "expected octal digit after '0o' prefix"

on line 174 (binary 0b) we peek a char and if it is not a digit with base `2` we return error "expected binary digit after '0b' prefix"

This approach has the added benefit of having clearer error messages